src/test/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrderingIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+
+    private static final String SAMPLE_NAME_KEY = "SN";
+    private static final String REVERSE_ORDERED_SAMPLE_MAP = "reverseOrdered.sample_map";
+    private static final String BADLY_SORTED1000_BATCH_SIZE50_VCF = "badlySorted1000-batch-size50.vcf";
+
+    private static File writeManyVCFs(int howMany) throws IOException {
+        final Map<String,File> nameToFileMap = new LinkedHashMap<>();
+        final String contig = "chr20";
+        final SAMSequenceDictionary dict = new SAMSequenceDictionary(
+                Collections.singletonList(new SAMSequenceRecord(contig, 64444167)));
+
+        final VCFFormatHeaderLine formatField = new VCFFormatHeaderLine(SAMPLE_NAME_KEY, 1, VCFHeaderLineType.String,
+                                                                        "the name of the sample this genotype came from");
+        final Set<VCFHeaderLine> headerLines = new HashSet<>();
+        headerLines.add(formatField);
+        headerLines.add(VCFStandardHeaderLines.getFormatLine("GT"));
+
+        for( int i = 0; i < howMany ; i++) {
+            final File out = File.createTempFile("tiny_", ".vcf");
+
+            try (final VariantContextWriter writer = GATKVariantContextUtils.createVCFWriter(out, dict, false, Options.INDEX_ON_THE_FLY)) {
+                final String sampleName = "Sample_" + String.valueOf(i);
+
+                final VCFHeader vcfHeader = new VCFHeader(headerLines, Collections.singleton(sampleName));
+                vcfHeader.setSequenceDictionary(dict);
+                writer.writeHeader(vcfHeader);
+                final Allele Aref = Allele.create("A", true);
+                final Allele C = Allele.create("C");
+                final List<Allele> alleles = Arrays.asList(Aref, C);
+                final VariantContext variant = new VariantContextBuilder("invented", contig, 100, 100, alleles)
+                        .genotypes(new GenotypeBuilder(sampleName, alleles).attribute(SAMPLE_NAME_KEY, sampleName).make())
+                        .make();
+                writer.add(variant);
+                nameToFileMap.put(sampleName, out);
+            }
+        }
+
+        return IOUtils.writeTempFile(
+                nameToFileMap.entrySet().stream()
+                        .map(pair -> pair.getKey() + "\t" + pair.getValue())
+                        .sorted(Comparator.reverseOrder())
+                        .collect(Collectors.joining("\n")), "mapping", ".sample_map");
+    }
+
+    @DataProvider
+    public Object[][] getBadInputs() {
+        return new Object[][]{
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile(BADLY_SORTED1000_BATCH_SIZE50_VCF), 10000}, //larger batch size than number of samples
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile(BADLY_SORTED1000_BATCH_SIZE50_VCF), 0}, // batch size 0
+                {getTestFile("mismatched.sample_map"), getTestFile(BADLY_SORTED1000_BATCH_SIZE50_VCF), 50} //larger batch size than number of samples
```

Comment:
larger -> smaller

Reply:
change comment to reflect the reason this test case triggers an error (the sample names in the input sample map are mismatched with those in the vcf)

Reply:
fixed fixed

Reply:
fixed, and fixed something wrong with the test as well.  Added a new subtype of UserException to disambiguate future failures